common/dynamicconfig/constants.go

```diff
@@ -1275,10 +1275,20 @@ these log lines can be noisy, we want to be able to turn on and sample selective
 		false,
 		`MatchingDropNonRetryableTasks states if we should drop matching tasks with Internal/Dataloss errors`,
 	)
+	MatchingMaxDeployments = NewNamespaceIntSetting(
+		"matching.maxDeployments",
+		1000,
+		`MatchingMaxDeployments represents the maximum number of worker deployments that can be registered in a single namespace`,
+	)
+	MatchingMaxVersionsInDeployment = NewNamespaceIntSetting(
+		"matching.maxVersionsInDeployment",
+		1000,
```

Comment:
I'm inclined to reduce this limit to say 100 for now until we get more confidence.